icu4c/source/data/buildtool/filtration.py

```diff
@@ -166,6 +171,118 @@ def _preprocess_file_filters(requests, config):
     return filters
 
 
-def _apply_resource_filters(old_requests, config):
+class ResourceFilterInfo(object):
+    def __init__(self, category):
+        self.category = category
+        self.filter_tmp_dir = "filters/%s" % category
+        self.input_files = None
+        self.filter_files = None
+        self.rules_by_file = None
+
+    def apply_to_requests(self, all_requests):
+        # Call this method only once per list of requests.
+        assert self.input_files is None
+        for request in all_requests:
+            if request.category != self.category:
+                continue
+            if not isinstance(request, AbstractExecutionRequest):
+                continue
+            if request.tool != IcuTool("genrb"):
+                continue
+            self._set_files(request.input_files)
+            # Add dependencies directly to dep_files
+            request.dep_files += self.filter_files
+            arg_str = "--filterDir {TMP_DIR}/%s" % self.filter_tmp_dir
+            request.args = "%s %s" % (arg_str, request.args)
+
+        # Make sure we found the target request
+        if self.input_files is None:
+            print("WARNING: Category not found: %s" % self.category, file=sys.stderr)
+            self.input_files = []
+            self.filter_files = []
+            self.rules_by_file = []
+
+    def _set_files(self, files):
+        if self.input_files is not None:
+            assert self.input_files == files
+            return
+        self.input_files = list(files)
+        self.filter_files = [
```

Comment:
Why assign, not append, here and below for self.rules_by_file?
Other than that inside the loop you have `request.dep_files += self.filter_files` but that could then be after the loop.

Reply:
self.filter_files and self.rules_by_file start out as None, and here in this method, they get initialized.  They will never get initialized twice thanks to the if statement and assertion above.

Reply:
Ok. Please add comments similar to your replies.